webrtc/RTCPeerConnection-helper.js

```diff
@@ -159,39 +159,90 @@ function test_never_resolve(testFunc, testName) {
   }, testName);
 }
 
+// Helper class to queue ICE candidates of a peer connection
+// Can be passed to `exchangeIceCandidates` which will automatically empty the queue.
+class IceCandidateQueue {
+  constructor(pc) {
+    this.pc = pc;
+    this._queue = [];
+    this._listener = (event) => this._handleIceCandidate(event);
+
+    // Queue ICE candidates
+    this.pc.addEventListener('icecandidate', this._listener);
+  }
+
+  // Unbind the event listener and return all queued candidates
+  disband() {
+    this.pc.removeEventListener('icecandidate', this._listener);
+    return this._queue;
+  }
+
+  _handleIceCandidate(event) {
+    this._queue.push(event.candidate);
+  }
+}
+
 // Helper function to exchange ice candidates between
 // two local peer connections
-function exchangeIceCandidates(pc1, pc2) {
-  // private function
-  function doExchange(localPc, remotePc) {
-    localPc.addEventListener('icecandidate', event => {
-      const { candidate } = event;
+function exchangeIceCandidates(pc1OrQueue, pc2OrQueue) {
```

Comment:
Removing *IceCandidateQueue* would mean putting *exchangeIceCandidates* back the way it was.